Tools/CISupport/ews-build/steps.py

```diff
@@ -1028,10 +1028,8 @@ def run(self):
             self.shell_command('git remote add {} {}{}.git || {}'.format(remote, GITHUB_URL, project, self.shell_exit_0())),
             ['git', 'remote', 'set-url', remote, '{}{}.git'.format(GITHUB_URL, project)],
             ['git', 'fetch', remote, '--prune'],
-            ['git', 'branch', '-f', pr_branch, 'remotes/{}/{}'.format(remote, pr_branch)],
-            ['git', 'checkout', pr_branch],
-            ['git', 'config', 'merge.changelog.driver', 'perl Tools/Scripts/resolve-ChangeLogs --merge-driver -c %O %A %B'],
-            ['git', 'rebase', rebase_target_hash, pr_branch],
+            ['git', 'checkout', '-b', pr_branch],
```

Comment:
Not a `git` expert, but I thought that we wanted to check out `main` here.

Reply:
We're leaning on the existing checkout state, which was configured by `CheckOutSource` and `CheckOutSpecificRevision`. That's going to be either the base branch for builders (which will be `main` most of the time) or the `ews_revision` as passed to the tester by the builder. Somewhat confusingly, `git checkout -b <branch name>` doesn't checkout the remote version of that branch into your current local checkout, it creates a local branch named `<branch name>` from the current checkout state. We probably don't have to do this, but I wanted to keep the local checkout "using" the PR branch, for `merge-queue`, in particular, it means we don't have to change anything.

Reply:
Ah right, not sure how I overlooked `-b`.

Certainly makes sense locally. It's icky that a new branch is created for an existing branch, can there be conflicts in any scenarios? I don't immediately see the code that deletes the local pr_branch, and I'm not sure if pushing it still works when it's created like this. I guess I'll leave reviewing to someone who understands the overall structure of this code better.

Reply:
> I don't immediately see the code that deletes the local pr_branch
Deleting all local branches happens here https://github.com/WebKit/WebKit/blob/6e160d801876a3b2752d4c39beb0cf542afaeb3f/Tools/CISupport/ews-build/steps.py#L4626

Reply:
>I'm not sure if pushing it still works when it's created like this.

merge-queue will force push to the branch, so this should "just work"

Reply:
>  >I'm not sure if pushing it still works when it's created like this.
> 
> 
> 
> merge-queue will force push to the branch, so this should "just work"

Right, and we have to do it this way to get a PR "merged" instead of closed, because the commit we're actually landing gets its message changed when we insert the reviewer and add the canonicalization link